system/pam_limits.py

```diff
@@ -191,11 +197,23 @@ def main():
                 nf.write(line)
                 continue
 
+            actual_value_unlimited = actual_value in ['unlimited', 'infinity', '-1']
+
             if use_max:
-                new_value = max(value, actual_value)
+                if value.isdigit() and actual_value.isdigit():
+                    new_value = max(value, actual_value)
```

Comment:
Not sure if value can be a two digit number but if so, this needs to convert to integers before comparing value and actual_value like this:
``` python
new_value = max(int(value), int(actual_value))
```
the reason is that something like ```max('10', '9')``` returns ```'9'```.

The same goes for the call to ```min(value, actual_value)``` below.